core/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -203,6 +203,29 @@ public ClusterBlockException indicesBlockedException(ClusterBlockLevel level, St
         return new ClusterBlockException(unmodifiableSet(blocks.collect(toSet())));
     }
 
+    public ClusterBlockException indicesAllowReleaseResources(String[] indices) {
+        boolean indexIsBlocked = false;
+        for (String index : indices) {
+            if (indexBlocked(ClusterBlockLevel.METADATA_WRITE, index)) {
+                indexIsBlocked = true;
+            }
+        }
+        if (globalBlocked(ClusterBlockLevel.METADATA_WRITE) == false && indexIsBlocked == false) {
```

Comment:
it seems `indexBlocked` already checks for the global blocks, so another check here is redundant. The only exception is if someone passes in an empty array but that is not the intended use of the method.